h2/src/main/org/h2/expression/function/Function.java

```diff
@@ -390,7 +390,7 @@
         addFunctionNotDeterministic("IDENTITY_VAL_LOCAL", IDENTITY,
                 0, Value.LONG);
         addFunctionNotDeterministic("LAST_INSERT_ID", IDENTITY,
-                0, Value.LONG);
+                VAR_ARGS, Value.LONG);
```

Comment:
You need a separate function number for `LAST_INSERT_ID` because after your changes it is different from other functions mapped to `IDENTITY`.

Reply:
Why LAST_INSERT_ID are in Functions, but not in FunctionsMySQL?

Reply:
Because it was implemented a long time ago much earlier than support for mode-specific functions.

Yes, it's better to remove the mapping for it from `Function` and move your implementation to `FunctionsMySQL`. We don't need this function in other modes.

You can use `SET MODE MySQL;` command in `last-insert-id.sql` to change the mode. It will be necessary, because functions from `FunctionsMySQL` are not available by default.